libs/core/src/lib/popover/popover-directive/popover.directive.ts

```diff
@@ -351,18 +357,18 @@ export class PopoverDirective implements OnInit, OnDestroy, OnChanges {
         }
     }
 
-    /** @hidden */
-    @HostListener('document:click', ['$event'])
-    clickHandler(event: MouseEvent): void {
-        if (this.containerRef &&
-            this.isOpen &&
-            this.closeOnOutsideClick &&
-            event.target !== this.elRef.nativeElement &&
-            !this.elRef.nativeElement.contains(event.target) &&
-            !this.containerRef.location.nativeElement.contains(event.target)) {
-            event.preventDefault();
-            event.stopPropagation();
-            this.close();
-        }
+    private _addListenerForOutsideClick(): void {
+         this.renderer.listen('document', 'click', (event: MouseEvent) => {
+             if (this.containerRef &&
```

Comment:
You can write it lie `if(shouldClosePopUp(event)){...}`. It's more readable for someone else who will take your code (event future yourself).